File: app/src/main/java/org/odddev/fantlab/core/layers/router/Router.java
package org.odddev.fantlab.core.layers.router;

import android.app.Activity;
import android.support.annotation.IdRes;

/**
 * @author kenrube
 * @since 28.09.16
 */

public class Router<T extends Activity> {

    private T activity;
    private @IdRes int containerId;

    public Router(T activity, @IdRes int containerId) {
        this.activity = activity;
        this.containerId = containerId;
    }

    protected T getActivity() {
        return activity;
    }

    public int getContainerId() {
        return containerId;
    }
}

File: app/src/main/java/org/odddev/fantlab/core/layers/router/Router.kt
package org.odddev.fantlab.core.layers.router

import android.app.Activity
import android.support.annotation.IdRes

/**
 * @author kenrube
 * *
 * @since 28.09.16
 */

open class Router<out T : Activity>(protected val activity: T, @IdRes val containerId: Int)
